### PR TITLE
Deprecate support for `ember-cli-qunit` and `ember-cli-mocha` when generating test blueprints

### PR DESCRIPTION
**Pull request by bertdeblock**
[Rendered RFC](https://github.com/bertdeblock/rfcs/blob/deprecate-support-for-ember-cli-qunit-and-ember-cli-mocha-when-generating-test-blueprints/text/0847-deprecate-support-for-ember-cli-qunit-and-ember-cli-mocha-when-generating-test-blueprints.md)

### CODE REVIEW

**Comment by bertdeblock** [comment]
It seems we also generate different test files depending on the `ember-mocha` version installed in the app:
https://github.com/emberjs/ember.js/blob/master/blueprints/test-framework-detector.js#L31-L41

Should we also deprecate support for `ember-mocha` versions below `v0.14.0`?

What's the status actually of `ember-mocha`? Development seems to have stalled.

**Comment by ef4** [comment]
This seems fine and uncontroversial.

**Comment by kategengler** [comment]
This was discussed in today's ember-cli meeting and we have decided to move it into Final Comment Period.

File: text/0847-deprecate-support-for-ember-cli-qunit-and-ember-cli-mocha-when-generating-test-blueprints.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+---
+stage: accepted
+start-date: 2022-08-26T00:00:00.000Z
+release-date:
+release-versions:
+teams: # delete teams that aren't relevant
+  - cli
+  - data
+  - framework
```
Comment (wagenet) [inline text/0847-deprecate-support-for-ember-cli-qunit-and-ember-cli-mocha-when-generating-test-blueprints.md:9]: This needs to be updated, right?

Comment (kategengler) [reply text/0847-deprecate-support-for-ember-cli-qunit-and-ember-cli-mocha-when-generating-test-blueprints.md:9]: I believe this list is correct in this case. There are blueprints in both data and cli that will be affected.

Comment (runspired) [reply text/0847-deprecate-support-for-ember-cli-qunit-and-ember-cli-mocha-when-generating-test-blueprints.md:9]: confirm, which is why I shipped it. Consider my review to be data's ack that blueprint changes are to be made.

Comment (wagenet) [reply text/0847-deprecate-support-for-ember-cli-qunit-and-ember-cli-mocha-when-generating-test-blueprints.md:9]: 👍

**Comment by kategengler** [comment]
The framework team discussed this today (with a large subset of ember-cli team) and we've decided to merge since no new discussion has happened since FCP.

**Pull request merged**